kafka-rest/src/main/java/io/confluent/kafkarest/ProducerPool.java

```diff
@@ -15,189 +15,25 @@
 
 package io.confluent.kafkarest;
 
-import io.confluent.kafka.schemaregistry.avro.AvroSchemaProvider;
-import io.confluent.kafka.schemaregistry.json.JsonSchemaProvider;
-import io.confluent.kafka.schemaregistry.protobuf.ProtobufSchemaProvider;
-import io.confluent.kafka.serializers.KafkaAvroSerializer;
-import io.confluent.kafka.serializers.KafkaJsonSerializer;
-import io.confluent.kafka.serializers.json.KafkaJsonSchemaSerializer;
-import io.confluent.kafka.serializers.protobuf.KafkaProtobufSerializer;
-import io.confluent.kafkarest.converters.AvroConverter;
-import io.confluent.kafkarest.converters.JsonSchemaConverter;
-import io.confluent.kafkarest.converters.ProtobufConverter;
-import io.confluent.kafkarest.entities.EmbeddedFormat;
-import io.confluent.kafkarest.entities.ProduceRequest;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Properties;
-import org.apache.kafka.clients.producer.KafkaProducer;
-import org.apache.kafka.common.serialization.ByteArraySerializer;
-import org.apache.kafka.common.serialization.Serializer;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import static java.util.Objects.requireNonNull;
+
+import org.apache.kafka.clients.producer.Producer;
 
 /**
- * Shared pool of Kafka producers used to send messages. The pool manages batched sends, tracking
- * all required acks for a batch and managing timeouts. Currently this pool only contains one
- * producer per serialization format (e.g. byte[], Avro).
+ * @deprecated This class only exists to satisfy the {@link KafkaRestContext} interface. It is soon
+ *             to be deleted. Access the producer directly, either via {@link
+ *             KafkaRestContext#getProducer()} or via injection of {@code Producer<byte[], byte[]>}.
```

Comment:
it seems to me like you can get rid of it now

Reply:
Ditto confluent-security-plugins.